apps/site/assets/ts/hooks/useStop.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from "react";
+import { fetchJsonOrThrow } from "../helpers/fetch-json";
+import { Stop } from "../__v3api";
+
+const fetchData = async (stopIdString: string): Promise<Stop> =>
+  fetchJsonOrThrow(`/api/stop/${stopIdString}`);
+
+const useStop = (stopId: string): Stop | null => {
+  const [stop, setStop] = useState<Stop | null>(null);
+
+  useEffect(() => {
+    fetchData(stopId).then(result => setStop(result));
+  }, [stopId]);
```

Comment:
In the interest of trying new approaches I was wondering if you'd like to try using an [alternate approach to data fetching](https://beta.reactjs.org/learn/synchronizing-with-effects#what-are-good-alternatives-to-data-fetching-in-effects) here. Notably, we already use one of the open-source libraries suggested by the linked post - [useSWR](https://swr.vercel.app), which we use successfully in the `useRealtime` hook. I think it'd be a really nice way to simplify our API-calling in general!